File: PartyApp/app/src/main/java/com/example/alex/partyapp/login/LoginFragment.java
package com.example.alex.partyapp.login;

import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.alex.partyapp.BaseFragment;
import com.example.alex.partyapp.BuildConfig;
import com.example.alex.partyapp.R;
import com.example.alex.partyapp.databinding.FragmentLoginBinding;
import com.example.alex.partyapp.di.App;

import javax.inject.Inject;

public class LoginFragment extends BaseFragment<FragmentLoginBinding> {

    @Inject
    LoginViewModel.Factory factory;
    private LoginViewModel viewModel;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_login, container, false);
        return binding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        App.getAppComponent(this.getContext()).inject(this);
        viewModel = ViewModelProviders.of(this, factory).get(LoginViewModel.class);
        viewModel.getStatus().observe(this, loginStatus -> {
            switch (loginStatus){
                case LOGGING:
                    binding.progressBar.setVisibility(View.VISIBLE);
                    break;
                case LOGGED_IN:
                    navigateToLoading();
                default:
                    binding.progressBar.setVisibility(View.INVISIBLE);
            }
        });
        viewModel.getError().observe(this, error -> {
            switch (error) {
                case FAILED:
                    showMessage(R.string.server_error);
                    break;
                case NETWORK_ERROR:
                    showMessage(R.string.network_error);
                    break;
                case AUTHORIZATION_FAILED:
                    showMessage(R.string.authorization_failed);
                    break;
            }
        });

        binding.buttonLogin.setOnClickListener(view -> viewModel.login(
                binding.editUsername.getText().toString(),
                binding.editPassword.getText().toString()));

        if (BuildConfig.DEBUG) {
            binding.editUsername.setText("tesonet");
            binding.editPassword.setText("partyanimal");
        }
    }
}

File: app/src/main/java/net/justinas/tesonetapp/withlib/view/LoginFragment.kt
package net.justinas.tesonetapp.withlib.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import net.justinas.tesonetapp.withlib.R
import net.justinas.tesonetapp.withlib.databinding.FragmentLoginBinding
import org.koin.androidx.viewmodel.ext.android.viewModel

class LoginFragment : Fragment() {

    private val viewModel: LoginViewModel by viewModel()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = FragmentLoginBinding.inflate(layoutInflater, container, false)
        binding.viewModel = viewModel
        binding.lifecycleOwner = viewLifecycleOwner
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setOnLoginGoNext()
    }

    private fun setOnLoginGoNext() {
        viewModel.successLogin.observe(viewLifecycleOwner, Observer {
            requireActivity().supportFragmentManager.beginTransaction()
                .setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(R.id.container, ListViewFragment(), "list")
                .addToBackStack("list")
                .commit()
        })
    }
}
